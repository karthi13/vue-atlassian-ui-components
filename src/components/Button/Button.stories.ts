import type { Meta, StoryObj } from '@storybook/vue3';

import Button from './Button.vue';
import ButtonStory from './stories/ButtonStory.vue';

// More on how to set up stories at: https://storybook.js.org/docs/writing-stories
const meta = {
  component: Button,
  // This component will have an automatically generated docsPage entry: https://storybook.js.org/docs/writing-docs/autodocs
  tags: ['autodocs'],
  argTypes: {
    appearance: {
      control: 'select',
      options: [
        'default',
        'primary',
        'warning',
        'danger',
        'link',
        'success',
        'subtle',
      ],
    },
    size: { control: 'select', options: ['sm', 'md', 'lg'] },
    outline: { control: 'boolean' },
    iconAfter: { control: 'boolean' },
    iconBefore: { control: 'boolean' },
    shape: { control: 'select', options: ['rounded', 'square', 'pill'] },
    nativeType: { control: 'select', options: ['button', 'submit', 'reset'] },
  },
  args: {
    appearance: 'primary',
    size: 'md',
    outline: false,
    iconAfter: false,
    iconBefore: false,
    shape: 'rounded',
    nativeType: 'button',
  },
} satisfies Meta<typeof Button>;

/*
 *ðŸ‘‡ Render functions are a framework specific feature to allow you control on how the component renders.
 * See https://storybook.js.org/docs/api/csf
 * to learn how to use render functions.
 */
type Story = StoryObj<typeof meta>;

export const Primary: Story = {
  render: () => ({
    components: { ButtonStory },
    template: '<ButtonStory />',
  }),
  decorators: [() => ({ template: '<story />' })],
};

export default meta;
