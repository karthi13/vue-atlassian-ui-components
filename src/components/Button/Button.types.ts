export enum BUTTON_APPEARANCE {
  PRIMARY = 'primary',
  WARNING = 'warning',
  DANGER = 'danger',
  SUCCESS = 'success',
  SUBTLE = 'subtle',
  DEFAULT = 'default',
}

export enum BUTTON_SIZE {
  SMALL = 'sm',
  MEDIUM = 'md',
  LARGE = 'lg',
}

export enum BUTTON_SHAPE {
  ROUNDED = 'rounded',
  SQUARE = 'square',
  PILL = 'pill',
}