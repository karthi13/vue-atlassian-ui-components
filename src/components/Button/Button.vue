<script setup lang="ts">
import { type ButtonHTMLAttributes } from 'vue';
import { twMerge } from 'tailwind-merge';
import { cva, type VariantProps } from 'class-variance-authority';

const button = cva(
  [
    'inline-flex place-items-center h-8 px-2.5 w-fit text-center disabled:cursor-not-allowed disabled:text-[#BFDBF847] disabled:bg-[#BCD6F00A]',
  ],
  {
    variants: {
      appearance: {
        default: [
          'bg-[#A1BDD914] hover:bg-[#A6C5E229] active:bg-[#BFDBF847] text-[#B6C2CF]',
        ],
        primary: [
          'bg-[#579DFF] hover:bg-[#85B8FF] active:bg-[#CCE0FF] text-[#1d2125]',
        ],
        subtle: [
          'bg-[#00000000] hover:bg-[#A1BDD914] active:bg-[#A6C5E229] text-[#B6C2CF]',
        ],
        warning: [
          'bg-[#f5cc47] hover:bg-[#E2B203] active:bg-[#CF9F02] text-[#1d2125]',
        ],
        danger: [
          'bg-[#f87268] hover:bg-[#FD9891] active:bg-[#FFD5D2] text-[#1d2125]',
        ],
        success: [
          'bg-[#47c978] hover:bg-[#2EBD6E] active:bg-[#1EBD6E] text-[#1d2125]',
        ],
      },
      size: {
        sm: ['font-medium tracking-normal leading-8 h-6 px-2 text-xs'],
        md: ['py-4 h-8 px-2.5 text-sm'],
        lg: ['text-lg py-6 px-3 leading-4'],
      },
      outline: {
        true: ['border bg-transparent hover:text-white'],
      },
      iconAfter: {
        true: ['flex-row-reverse gap-x-1'],
      },
      iconBefore: {
        true: ['flex-row gap-x-1'],
      },
      shape: {
        rounded: ['rounded'],
        square: ['rounded-none'],
        pill: ['rounded-full'],
      },
    },
    compoundVariants: [
      {
        appearance: 'primary',
        outline: true,
        class: 'border-btn-primary text-btn-primary hover:bg-btn-primary ',
      },
      {
        appearance: 'warning',
        outline: true,
        class: 'border-btn-warning text-btn-warning hover:bg-btn-warning',
      },
      {
        appearance: 'danger',
        outline: true,
        class: 'border-btn-danger text-btn-danger hover:bg-btn-danger',
      },
      {
        appearance: 'success',
        size: ['sm', 'md', 'lg'],
        outline: true,
        class: 'border-green-500 text-green-500 hover:bg-green-500',
      },
    ],
    defaultVariants: {
      appearance: 'default',
      size: 'md',
      shape: 'rounded',
      outline: false,
      iconAfter: false,
      iconBefore: false,
    },
  },
);
type ButtonProps = VariantProps<typeof button>;

const props = withDefaults(
  defineProps<{
    appearance?: ButtonProps['appearance'];
    size?: ButtonProps['size'];
    shape?: ButtonProps['shape'];
    outline?: ButtonProps['outline'];
    nativeType?: ButtonHTMLAttributes['type'];
    iconAfter: ButtonProps['iconAfter'];
    iconBefore: ButtonProps['iconBefore'];
  }>(),
  {
    nativeType: 'button',
    iconAfter: false,
    iconBefore: false,
  },
);

console.log(props);
// import { twMerge } from "tailwind-merge";
// import { cva, type VariantProps } from "class-variance-authority";

// const defaultButtonClasses = `h-8 px-2.5 rounded place-items-center focus:outline-none box-border
// inline-flex font-medium text-sm text-center w-fit disabled:cursor-not-allowed disabled:text-[#BFDBF847]
// disabled:bg-[#BCD6F00A]`;

// const buttonCVA = cva(defaultButtonClasses, {
//   variants: {
//     appearance: {
//       default: "bg-btn-neutral hover:bg-btn-neutral-hover active:bg-btn-neutral-pressed",
//       primary: "bg-btn-primary hover:bg-btn-primary-hover active:bg-btn-primary-active",
//       warning: "bg-btn-warning hover:bg-btn-warning-hover active:bg-btn-warning-active",
//       danger: "bg-btn-danger hover:bg-btn-danger-hover active:bg-btn-danger-active",
//       link: "bg-btn-subtle text-link-primary hover:underline underline-offset-2",
//       subtle: "bg-btn-subtle active:bg-btn-subtle-active hover:bg-btn-subtle-hover",
//       "subtle-link": "bg-btn-subtle hover:underline underline-offset-2",
//     },
//     size: {
//       sm: "h-6 px-2 rounded text-xs",
//       md: "h-8 px-2.5 rounded text-sm",
//       lg: "h-12 px-3 rounded text-lg align-center text-center",
//     },
//     edge: {
//       rounded: "rounded",
//       square: "rounded-none",
//       pill: "rounded-full",
//     },
//     outline: {
//       true: "border ",
//     },
//   },
//   compoundVariants: [
//     { appearance: "default", outline: true, class: "border-btn-primary text-btn-primary" },
//     { appearance: "primary", outline: true, class: "border-btn-warning text-btn-warning" },
//     { appearance: "warning", outline: true, class: "border-btn-danger text-btn-danger" },
//     { appearance: "danger", outline: true, class: "border-green-500 text-green-500" },
//   ],
// });

// type ButtonProps = VariantProps<typeof buttonCVA>;

//   withDefaults(
//     defineProps<{
//       appearance: ButtonProps["appearance"];
//       size: ButtonProps["size"];
//       edge: ButtonProps["edge"];
//       outline: ButtonProps["outline"];
//       nativeType: {
//         type: string,
//         default: "button",
//       }
//       iconAfter: {
//         type: boolean,
//         default: false,
//       },
//       iconBefore: {
//         type: boolean,
//         default: false,
//       },
//     }>(),
//     {
//       appearance: "primary",
//       size: "md",
//       edge: "rounded",
//       outline: false,
//     },
//     );

// type ButtonAppearance = "default" | "primary" | "warning" | "danger" | "link" | "subtle" | "subtle-link";
// type ButtonSize = "sm" | "md" | "lg";
// type ButtonNativeType = "button" | "submit" | "reset" | undefined;

// const props = defineProps({
//   appearance: {
//     type: String as () => ButtonAppearance,
//     default: "default",
//     validator: (value: string) => {
//       return ["default", "primary", "warning", "danger", "link", "subtle", "subtle-link"].includes(value);
//     },
//   },
//   edge: {
//     type: String,
//     default: "rounded",
//     validator: (value: string) => {
//       return ["rounded", "square", "pill"].includes(value);
//     },
//   },
//   size: {
//     type: String as () => ButtonSize,
//     default: "md",
//     validator: (value: string) => {
//       return ["sm", "md", "lg"].includes(value);
//     },
//   },
//   isLoading: {
//     type: Boolean,
//     default: false,
//   },
//   iconAfter: {
//     type: Boolean,
//     default: false,
//   },
//   iconBefore: {
//     type: Boolean,
//     default: false,
//   },
//   href: {
//     type: String,
//     default: "",
//   },
//   outline: {
//     type: Boolean,
//     default: false,
//   },
//   isDisabled: {
//     type: Boolean,
//     default: false,
//   },
//   isSelected: {
//     type: Boolean,
//     default: false,
//   },
//   target: {
//     type: String,
//     default: "_top",
//     validator: (value: string) => {
//       return ["_top", "_blank", "_self", "_parent"].includes(value);
//     },
//   },
//   nativeType: {
//     type: String as () => ButtonNativeType,
//     default: "button",
//     validator: (value: string) => {
//       return ["button", "submit", "reset"].includes(value);
//     },
//   },
// });

// const buttonAppearance = computed(() => {
//   if (props.outline) {
//     switch(props.appearance) {
//       case "primary":
//         return "border border-btn-primary text-btn-primary";
//       case "warning":
//         return "border border-btn-warning text-btn-warning";
//       case "danger":
//         return "border border-btn-danger text-btn-danger";
//       default:
//         return "border border-green-500 text-green-500";
//     }
//   }
//   switch(props.appearance) {
//     case "primary":
//       return "bg-btn-primary hover:bg-btn-primary-hover active:bg-btn-primary-active";
//     case "warning":
//       return "bg-btn-warning hover:bg-btn-warning-hover active:bg-btn-warning-active";
//     case "danger":
//       return "bg-btn-danger hover:bg-btn-danger-hover active:bg-btn-danger-active";
//     case "link":
//       return "bg-btn-subtle text-link-primary hover:underline underline-offset-2";
//     case "subtle":
//       return "bg-btn-subtle active:bg-btn-subtle-active hover:bg-btn-subtle-hover";
//     case "subtle-link":
//       return "bg-btn-subtle hover:underline underline-offset-2";
//     default:
//       return "bg-btn-neutral hover:bg-btn-neutral-hover active:bg-btn-neutral-pressed";
//   }
// });

// const buttonSize = computed(() => {
//   switch(props.size) {
//     case "sm":
//       return "h-6 px-2 rounded text-xs";
//     case "lg":
//       return "h-12 px-3 rounded text-lg align-center text-center";
//     case "md":
//     default:
//       return "h-8 px-2.5 rounded text-sm";
//   }
// });

// const buttonText = computed(() => {
//   if (props.outline) {
//     switch(props.appearance) {
//       case "primary":
//         return "text-btn-primary leading-8";
//       case "warning":
//         return "text-btn-warning leading-8";
//       case "danger":
//         return "text-btn-danger leading-8";
//       default:
//         return "text-green-500 leading-8";
//     }
//   }
//   switch(props.appearance) {
//     case "primary":
//     case "warning":
//     case "danger":
//       return "text-inverse leading-8";
//     case "link":
//       return "text-link-primary active:text-link-primary-active active:underline leading-8";
//     case "subtle-link":
//         return "text-subtle hover:text-subtle active:text-default leading-8";
//     case "subtle":
//     default:
//       return "text-default leading-8";
//   }
// });
</script>

<template>
  <button
    :class="
      twMerge(
        button({ appearance, size, outline, shape, iconAfter, iconBefore }),
      )
    "
    :type="nativeType"
    tabindex="0">
    <slot />
  </button>
</template>
