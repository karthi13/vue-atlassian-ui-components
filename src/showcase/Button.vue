<template>
  <div class="flex flex-col w-full gap-2">
    <div class="flex flex-col gap-2">
      <Button>Default button</Button>
      <Button appearance="primary"> Primary button </Button>
      <Button appearance="subtle"> Subtle button </Button>
      <Button appearance="warning"> Warning button </Button>
      <Button appearance="danger"> Danger button </Button>
      <Button appearance="link"> Link button </Button>
      <Button appearance="subtle-link"> Subtle ling button </Button>
    </div>
    <hr />
    <div class="flex flex-col gap-2">
      <Button appearance="primary" is-disabled>
        Primary disabled button
      </Button>
    </div>
    <hr />
    <div class="flex flex-col gap-2">
      <Button appearance="primary" size="sm"> Small Button </Button>
      <Button appearance="primary" size="md"> Small Button </Button>
      <Button appearance="primary" size="lg"> Small Button </Button>
    </div>
    <hr />
    <div class="flex flex-col gap-2">
      <Button appearance="primary" size="lg" outline>
        Primary Outline button
      </Button>
      <Button size="sm" outline> Default Outline button </Button>
      <Button appearance="warning" size="md" outline>
        Warning Outline button
      </Button>
      <Button appearance="danger" size="sm" outline>
        Danger Outline button
      </Button>
    </div>
    <hr />
    <div class="flex flex-col gap-2">
      <Button appearance="primary" edge="rounded"> Primary button </Button>
      <Button appearance="danger" edge="square"> Subtle button </Button>
      <Button appearance="warning" edge="pill"> Warning button </Button>
    </div>
    <hr />
  </div>
</template>

<script setup lang="ts">
import Button from '../components/Button/Button.vue';
</script>

