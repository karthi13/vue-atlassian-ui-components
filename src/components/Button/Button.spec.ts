import { h } from 'vue'
import { render, screen } from "@testing-library/vue"
import Button from "./Button.vue"
import { BUTTON_APPEARANCE, BUTTON_SIZE, BUTTON_SHAPE } from './Button.types'

const BUTTON_TEXT = "Click me";

describe.concurrent('Button appearance', () => {
  it('default button', async () => {
    const buttonWrapper = render(Button, {
      slots: { default: BUTTON_TEXT }
    })
    const button = await screen.findByText(BUTTON_TEXT)
    expect(button.innerHTML).toBe(BUTTON_TEXT)
    expect(buttonWrapper.html()).toMatchSnapshot()
  });

  it('primary button', async () => {
    const buttonWrapper = render(Button, {
      props: { appearance: BUTTON_APPEARANCE.PRIMARY },
      slots: { default: BUTTON_TEXT }
    })
    expect(buttonWrapper.html()).toMatchSnapshot()
  });

  it('subtle button', async () => {
    const buttonWrapper = render(Button, {
      props: { appearance: BUTTON_APPEARANCE.SUBTLE },
      slots: { default: BUTTON_TEXT }
    })
    expect(buttonWrapper.html()).toMatchSnapshot()
  });

  it('warning button', async () => {
    const buttonWrapper = render(Button, {
      props: { appearance: BUTTON_APPEARANCE.WARNING },
      slots: { default: BUTTON_TEXT }
    })
    expect(buttonWrapper.html()).toMatchSnapshot()
  });

  it('danger button', async () => {
    const buttonWrapper = render(Button, {
      props: { appearance: BUTTON_APPEARANCE.DANGER },
      slots: { default: BUTTON_TEXT }
    })
    expect(buttonWrapper.html()).toMatchSnapshot()
  });

  it('success button', async () => {
    const buttonWrapper = render(Button, {
      props: { appearance: BUTTON_APPEARANCE.SUCCESS },
      slots: { default: BUTTON_TEXT }
    })
    expect(buttonWrapper.html()).toMatchSnapshot()
  });
});

describe('Button size', () => {
  it('small button', async () => {
    const buttonWrapper = render(Button, {
      props: { size: BUTTON_SIZE.SMALL },
      slots: { default: BUTTON_TEXT }
    })
    expect(buttonWrapper.html()).toMatchSnapshot()
  });

  it('medium button', async () => {
    const buttonWrapper = render(Button, {
      props: { size: BUTTON_SIZE.MEDIUM },
      slots: { default: BUTTON_TEXT }
    })
    expect(buttonWrapper.html()).toMatchSnapshot()
  });

  it('large button', async () => {
    const buttonWrapper = render(Button, {
      props: { size: BUTTON_SIZE.LARGE },
      slots: { default: BUTTON_TEXT }
    })
    expect(buttonWrapper.html()).toMatchSnapshot()
  });
});

describe('Button outline', () => {
  it('default button', async () => {
    const buttonWrapper = render(Button, {
      props: { outline: true },
      slots: { default: BUTTON_TEXT }
    })
    expect(buttonWrapper.html()).toMatchSnapshot()
  });

  it('primary button', async () => {
    const buttonWrapper = render(Button, {
      props: { appearance: BUTTON_APPEARANCE.PRIMARY, outline: true },
      slots: { default: BUTTON_TEXT }
    })
    expect(buttonWrapper.html()).toMatchSnapshot()
  });

  it('subtle button', async () => {
    const buttonWrapper = render(Button, {
      props: { appearance: BUTTON_APPEARANCE.SUBTLE, outline: true },
      slots: { default: BUTTON_TEXT }
    })
    expect(buttonWrapper.html()).toMatchSnapshot()
  });

  it('warning button', async () => {
    const buttonWrapper = render(Button, {
      props: { appearance: BUTTON_APPEARANCE.WARNING, outline: true },
      slots: { default: BUTTON_TEXT }
    })
    expect(buttonWrapper.html()).toMatchSnapshot()
  });

  it('danger button', async () => {
    const buttonWrapper = render(Button, {
      props: { appearance: BUTTON_APPEARANCE.DANGER, outline: true },
      slots: { default: BUTTON_TEXT }
    })
    expect(buttonWrapper.html()).toMatchSnapshot()
  });

  it('success button', async () => {
    const buttonWrapper = render(Button, {
      props: { appearance: BUTTON_APPEARANCE.SUCCESS, outline: true },
      slots: { default: BUTTON_TEXT }
    })
    expect(buttonWrapper.html()).toMatchSnapshot()
  });
});

describe('Button shape', () => {
  it('rounded button', async () => {
    const buttonWrapper = render(Button, {
      props: { appearance: BUTTON_APPEARANCE.PRIMARY, shape: BUTTON_SHAPE.ROUNDED },
      slots: { default: BUTTON_TEXT }
    })
    expect(buttonWrapper.html()).toMatchSnapshot()
  });

  it('square button', async () => {
    const buttonWrapper = render(Button, {
      props: { appearance: BUTTON_APPEARANCE.PRIMARY, shape: BUTTON_SHAPE.SQUARE },
      slots: { default: BUTTON_TEXT }
    })
    expect(buttonWrapper.html()).toMatchSnapshot()
  });

  it('pill button', async () => {
    const buttonWrapper = render(Button, {
      props: { appearance: BUTTON_APPEARANCE.PRIMARY, shape: BUTTON_SHAPE.PILL },
      slots: { default: BUTTON_TEXT }
    })
    expect(buttonWrapper.html()).toMatchSnapshot()
  });
});

describe('Button with icons', () => {
  it('Icon before button', async () => {
    const buttonWrapper = render(Button, {
      props: { appearance: BUTTON_APPEARANCE.PRIMARY },
      slots: { default: [ h('email-icon', ''), ' ' , BUTTON_TEXT ] }
    })
    expect(buttonWrapper.html()).toMatchSnapshot()
  });

  it('Icon after button', async () => {
    const buttonWrapper = render(Button, {
      props: { appearance: BUTTON_APPEARANCE.PRIMARY },
      slots: { default: [ BUTTON_TEXT, ' ' , h('email-icon', '') ] }
    })
    expect(buttonWrapper.html()).toMatchSnapshot()
  });

  it('Icon only button', async () => {
    const buttonWrapper = render(Button, {
      props: { appearance: BUTTON_APPEARANCE.PRIMARY, shape: BUTTON_SHAPE.PILL },
      slots: { default: h('email-icon', '') }
    })
    expect(buttonWrapper.html()).toMatchSnapshot()
  });
});