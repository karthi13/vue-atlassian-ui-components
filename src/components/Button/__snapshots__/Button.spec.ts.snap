// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button appearance > danger button 1`] = `"<button class="inline-flex place-items-center w-fit text-center disabled:cursor-not-allowed disabled:text-[#BFDBF847] disabled:bg-[#BCD6F00A] bg-[#579DFF] hover:bg-[#85B8FF] active:bg-[#CCE0FF] text-[#1d2125] py-4 h-8 px-2.5 text-sm rounded" type="button" tabindex="0">Click me</button>"`;

exports[`Button appearance > danger button 2`] = `"<button class="inline-flex place-items-center w-fit text-center disabled:cursor-not-allowed disabled:text-[#BFDBF847] disabled:bg-[#BCD6F00A] bg-[#00000000] hover:bg-[#A1BDD914] active:bg-[#A6C5E229] text-[#B6C2CF] py-4 h-8 px-2.5 text-sm rounded" type="button" tabindex="0">Click me</button>"`;

exports[`Button appearance > danger button 3`] = `"<button class="inline-flex place-items-center w-fit text-center disabled:cursor-not-allowed disabled:text-[#BFDBF847] disabled:bg-[#BCD6F00A] bg-[#f5cc47] hover:bg-[#E2B203] active:bg-[#CF9F02] text-[#1d2125] py-4 h-8 px-2.5 text-sm rounded" type="button" tabindex="0">Click me</button>"`;

exports[`Button appearance > danger button 4`] = `"<button class="inline-flex place-items-center w-fit text-center disabled:cursor-not-allowed disabled:text-[#BFDBF847] disabled:bg-[#BCD6F00A] bg-[#f87268] hover:bg-[#FD9891] active:bg-[#FFD5D2] text-[#1d2125] py-4 h-8 px-2.5 text-sm rounded" type="button" tabindex="0">Click me</button>"`;

exports[`Button appearance > danger button 5`] = `"<button class="inline-flex place-items-center w-fit text-center disabled:cursor-not-allowed disabled:text-[#BFDBF847] disabled:bg-[#BCD6F00A] bg-[#A1BDD914] hover:bg-[#A6C5E229] active:bg-[#BFDBF847] text-[#B6C2CF] py-4 h-8 px-2.5 text-sm rounded" type="button" tabindex="0">Click me</button>"`;

exports[`Button appearance > success button 1`] = `"<button class="inline-flex place-items-center w-fit text-center disabled:cursor-not-allowed disabled:text-[#BFDBF847] disabled:bg-[#BCD6F00A] bg-[#47c978] hover:bg-[#2EBD6E] active:bg-[#1EBD6E] text-[#1d2125] py-4 h-8 px-2.5 text-sm rounded" type="button" tabindex="0">Click me</button>"`;

exports[`Button outline > danger button 1`] = `"<button class="inline-flex place-items-center w-fit text-center disabled:cursor-not-allowed disabled:text-[#BFDBF847] disabled:bg-[#BCD6F00A] active:bg-[#FFD5D2] py-4 h-8 px-2.5 text-sm border bg-transparent hover:text-white rounded border-btn-danger text-btn-danger hover:bg-btn-danger" type="button" tabindex="0">Click me</button>"`;

exports[`Button outline > default button 1`] = `"<button class="inline-flex place-items-center w-fit text-center disabled:cursor-not-allowed disabled:text-[#BFDBF847] disabled:bg-[#BCD6F00A] hover:bg-[#A6C5E229] active:bg-[#BFDBF847] text-[#B6C2CF] py-4 h-8 px-2.5 text-sm border bg-transparent hover:text-white rounded" type="button" tabindex="0">Click me</button>"`;

exports[`Button outline > primary button 1`] = `"<button class="inline-flex place-items-center w-fit text-center disabled:cursor-not-allowed disabled:text-[#BFDBF847] disabled:bg-[#BCD6F00A] active:bg-[#CCE0FF] py-4 h-8 px-2.5 text-sm border bg-transparent hover:text-white rounded border-btn-primary text-btn-primary hover:bg-btn-primary" type="button" tabindex="0">Click me</button>"`;

exports[`Button outline > subtle button 1`] = `"<button class="inline-flex place-items-center w-fit text-center disabled:cursor-not-allowed disabled:text-[#BFDBF847] disabled:bg-[#BCD6F00A] hover:bg-[#A1BDD914] active:bg-[#A6C5E229] text-[#B6C2CF] py-4 h-8 px-2.5 text-sm border bg-transparent hover:text-white rounded" type="button" tabindex="0">Click me</button>"`;

exports[`Button outline > success button 1`] = `"<button class="inline-flex place-items-center w-fit text-center disabled:cursor-not-allowed disabled:text-[#BFDBF847] disabled:bg-[#BCD6F00A] active:bg-[#1EBD6E] py-4 h-8 px-2.5 text-sm border bg-transparent hover:text-white rounded border-green-500 text-green-500 hover:bg-green-500" type="button" tabindex="0">Click me</button>"`;

exports[`Button outline > warning button 1`] = `"<button class="inline-flex place-items-center w-fit text-center disabled:cursor-not-allowed disabled:text-[#BFDBF847] disabled:bg-[#BCD6F00A] active:bg-[#CF9F02] py-4 h-8 px-2.5 text-sm border bg-transparent hover:text-white rounded border-btn-warning text-btn-warning hover:bg-btn-warning" type="button" tabindex="0">Click me</button>"`;

exports[`Button shape > pill button 1`] = `"<button class="inline-flex place-items-center w-fit text-center disabled:cursor-not-allowed disabled:text-[#BFDBF847] disabled:bg-[#BCD6F00A] bg-[#579DFF] hover:bg-[#85B8FF] active:bg-[#CCE0FF] text-[#1d2125] py-4 h-8 px-2.5 text-sm rounded-full" type="button" tabindex="0">Click me</button>"`;

exports[`Button shape > rounded button 1`] = `"<button class="inline-flex place-items-center w-fit text-center disabled:cursor-not-allowed disabled:text-[#BFDBF847] disabled:bg-[#BCD6F00A] bg-[#579DFF] hover:bg-[#85B8FF] active:bg-[#CCE0FF] text-[#1d2125] py-4 h-8 px-2.5 text-sm rounded" type="button" tabindex="0">Click me</button>"`;

exports[`Button shape > square button 1`] = `"<button class="inline-flex place-items-center w-fit text-center disabled:cursor-not-allowed disabled:text-[#BFDBF847] disabled:bg-[#BCD6F00A] bg-[#579DFF] hover:bg-[#85B8FF] active:bg-[#CCE0FF] text-[#1d2125] py-4 h-8 px-2.5 text-sm rounded-none" type="button" tabindex="0">Click me</button>"`;

exports[`Button size > large button 1`] = `"<button class="inline-flex place-items-center h-8 w-fit text-center disabled:cursor-not-allowed disabled:text-[#BFDBF847] disabled:bg-[#BCD6F00A] bg-[#A1BDD914] hover:bg-[#A6C5E229] active:bg-[#BFDBF847] text-[#B6C2CF] text-lg py-6 px-3 leading-4 rounded" type="button" tabindex="0">Click me</button>"`;

exports[`Button size > medium button 1`] = `"<button class="inline-flex place-items-center w-fit text-center disabled:cursor-not-allowed disabled:text-[#BFDBF847] disabled:bg-[#BCD6F00A] bg-[#A1BDD914] hover:bg-[#A6C5E229] active:bg-[#BFDBF847] text-[#B6C2CF] py-4 h-8 px-2.5 text-sm rounded" type="button" tabindex="0">Click me</button>"`;

exports[`Button size > small button 1`] = `"<button class="inline-flex place-items-center w-fit text-center disabled:cursor-not-allowed disabled:text-[#BFDBF847] disabled:bg-[#BCD6F00A] bg-[#A1BDD914] hover:bg-[#A6C5E229] active:bg-[#BFDBF847] text-[#B6C2CF] font-medium tracking-normal h-6 px-2 text-xs rounded" type="button" tabindex="0">Click me</button>"`;

exports[`Button with icons > Icon after button 1`] = `"<button class="inline-flex place-items-center w-fit text-center disabled:cursor-not-allowed disabled:text-[#BFDBF847] disabled:bg-[#BCD6F00A] bg-[#579DFF] hover:bg-[#85B8FF] active:bg-[#CCE0FF] text-[#1d2125] py-4 h-8 px-2.5 text-sm rounded" type="button" tabindex="0">Click me<email-icon></email-icon></button>"`;

exports[`Button with icons > Icon before button 1`] = `
"<button class="inline-flex place-items-center w-fit text-center disabled:cursor-not-allowed disabled:text-[#BFDBF847] disabled:bg-[#BCD6F00A] bg-[#579DFF] hover:bg-[#85B8FF] active:bg-[#CCE0FF] text-[#1d2125] py-4 h-8 px-2.5 text-sm rounded" type="button" tabindex="0">
  <email-icon></email-icon>Click me
</button>"
`;

exports[`Button with icons > Icon only button 1`] = `
"<button class="inline-flex place-items-center w-fit text-center disabled:cursor-not-allowed disabled:text-[#BFDBF847] disabled:bg-[#BCD6F00A] bg-[#579DFF] hover:bg-[#85B8FF] active:bg-[#CCE0FF] text-[#1d2125] py-4 h-8 px-2.5 text-sm rounded-full" type="button" tabindex="0">
  <email-icon></email-icon>
</button>"
`;
